runtime/commonMain/src/kotlinx/serialization/internal/CollectionSerializers.kt

```diff
@@ -268,10 +278,12 @@ class LinkedHashMapSerializer<K, V>(kSerializer: KSerializer<K>, vSerializer: KS
     override fun LinkedHashMap<K, V>.insertKeyValuePair(index: Int, key: K, value: V) = set(key, value)
 }
 
-class HashMapSerializer<K, V>(kSerializer: KSerializer<K>, vSerializer: KSerializer<V>) :
-    MapLikeSerializer<K, V, Map<K, V>, HashMap<K, V>>(kSerializer, vSerializer) {
-    override val descriptor = HashMapClassDesc(kSerializer.descriptor, vSerializer.descriptor)
+@InternalSerializationApi // Use kotlinx.serialization.builtins.MapSerializer() instead
+class HashMapSerializer<K, V>(
```

Comment:
Add `public` everywhere (in this or following PR)

Reply:
In the follow up with interfaces